portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,36 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Checking login status. 
+    UserService userService = UserServiceFactory.getUserService();
+    
+    if (userService.isUserLoggedIn()) {
+      String email = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterLogOut = "/login";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterLogOut);
+
+      response.getWriter().println("<p>Hello " + email + "!</p>");
+      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+
+    } else {
+      String urlToRedirectToAfterLogIn = "/data";
```

Comment:
Did you push your change? looks like this is still pointing to "/data"

Reply:
Oh thought I had that's strange, will do again :)

Reply:
Hi Travis! It shows as if it is already updated to me, could you check again? (the change was made in the "fixing log-in redirect" commit)